restapi/stories.go

```diff
@@ -130,16 +130,16 @@ func (api api) EditStoryByID(w http.ResponseWriter, r *http.Request) render.Rend
 			return rest.ErrInvalidRequest(api.logger, msg, err)
 		}*/
 
+	imageURL := story.ImageURL
+
 	file, h, err := r.FormFile("image")
 
-	if err != nil {
-		return rest.ErrInvalidRequest(api.logger, "Error getting the image.", err)
-	}
-	imageURL, err := api.uploadImageTos3(file, h.Size, h.Filename)
-	if err != nil {
-		return rest.ErrInvalidRequest(api.logger, "Error uploading the image to s3.", err)
+	if err == nil {
+		imageURL, err = api.uploadImageTos3(file, h.Size, h.Filename)
+		if err != nil {
+			return rest.ErrInvalidRequest(api.logger, "Error uploading the image to s3.", err)
+		}
 	}
-	defer file.Close()
```

Comment:
don't we still need to close the file if there was an image?

Reply:
this actually defers the closing of file. We dont need this cuz we are not really writing to the file, so we can close it.